tensorflow/contrib/mpi_collectives/BUILD

```diff
@@ -33,26 +21,94 @@ tf_proto_library_cc(
     ],
 )
 
-load("//tensorflow:tensorflow.bzl", "tf_custom_op_library")
-load("//tensorflow:tensorflow.bzl", "tf_py_test")
+load(
+    "//tensorflow:tensorflow.bzl",
+    "tf_custom_op_py_library",
+    "tf_custom_op_library",
+    "tf_gen_op_wrapper_py",
+    "tf_gen_op_libs",
+    "tf_kernel_library",
+    "tf_py_test",
+)
 
 tf_custom_op_library(
-    name = "mpi_collectives.so",
+    name = "python/ops/_mpi_ops.so",
     srcs = [
-        "mpi_ops.cc",
-        "ring.cc",
-        "ring.h",
+        "kernels/mpi_ops.cc",
+        "kernels/ring.cc",
+        "kernels/ring.h",
+        "ops/mpi_ops.cc",
     ],
     gpu_srcs = [
-        "ring.cu.cc",
-        "ring.h",
+        "kernels/ring.cu.cc",
+        "kernels/ring.h",
     ],
     deps = [
         ":mpi_message_proto_cc",
+        "//tensorflow/core:lib_internal",
```

Comment:
So this is (only) to pick up tf_additional_mpi_lib_defines()? Would it help if we added a defines or a copts field to [tf_custom_op_library](https://github.com/tensorflow/tensorflow/blob/d409f79cb29df6091cb3ee2c80f6b2bd826950e5/tensorflow/tensorflow.bzl#L1218)? Otherwise I'm fine with this, or I suppose you could create a separate `cc_library` rule with just the `defines=` if you wanted to avoid depending on the whole `lib_internal` rule.

One advantage of having your own defines= rule would be protection from people (e.g. me) ham-fistedly removing the define you're relying on from lib_internal. This was the main (only?) issue after the RTLD_GLOBAL removal, correct?

Reply:
Right, `lib_internal` includes `tf_additional_mpi_lib_defines()` for the preprocessor define `TENSORFLOW_USE_MPI`. This might have been the only issue after RTLD_GLOBAL removal.

Per your suggestion, I moved the defines over to a `cc_library` target within the `mpi_collectives` BUILD file. That works nicely. Thanks!